### ISSUE TITLE
Incorrect errors when subclassing `TypedDict` multiple times

### ISSUE THREAD

**Issue by Zomatree** [microsoft/pyright#5550]
**Describe the bug**
When subclassing `TypedDict` itself along with another subclass of `TypedDict` pyright emits `"Cannot create consistent method ordering"` when its valid at runtime.

Along with this subclassing `TypedDict` multiple times also emits `"Duplicate base class not allowed"` when its valid at runtime.

**To Reproduce**
```python
from typing import TypedDict


class Foo(TypedDict, TypedDict):
    pass

class Other(TypedDict, Foo):
    pass

class Another(Other, TypedDict):
    pass
```

```
pyright.py:4:22 - error: Duplicate base class not allowed (reportGeneralTypeIssues)
pyright.py:7:7 - error: Cannot create consistent method ordering
```

**Expected behavior**
No errors should be emitted.

**Code or Screenshots**
![Code_-_Insiders_FnblfnQJ8B](https://github.com/microsoft/pyright/assets/39768508/d111ac92-541c-4b9e-bd33-a27a5c4e0963)
If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**VS Code extension or command-line**
`pyright 1.1.318`

**Comment by erictraut** [comment]
The runtime must have a special case specifically for `TypedDict` because duplicate base classes in general produce an error at runtime. Mypy reports the same error with the above code. I don't think it's worth special-casing this in type checkers. The solution is to remove the redundant base classes in your code.
